File: backend/data_structures/source.py
"""
Source definitions for tournament data origins.
"""
from enum import StrEnum

class Source(StrEnum):
    LONGSHANKS = "longshanks"
    LISTFORTRESS = "listfortress"
    ROLLBETTER = "rollbetter"
    UNKNOWN = "unknown"

    @property
    def label(self) -> str:
        """Human-readable source name."""
        match self:
            case Source.LONGSHANKS: return "Longshanks"
            case Source.LISTFORTRESS: return "ListFortress"
            case Source.ROLLBETTER: return "Rollbetter"
            case _: return "Unknown"

File: frontend/src/lib/data/source.ts
export const SOURCE_LABELS: Record<string, string> = {
    listfortress: 'ListFortress',
    longshanks: 'Longshanks',
    rollbetter: 'Rollbetter',
    unknown: 'Unknown'
};

export function getSourceLabel(source: string): string {
    return SOURCE_LABELS[source.toLowerCase()] ?? source;
}
